### PR TITLE
Issues11+12 fix

### PR DESCRIPTION
**Pull request by jond01**
Fixed issues #11 and #12 + use 'SliceThickness' or 'SpacingBetweenSlices' as slice_spacing (in the case of a single slice dataset only if exists)

### CODE REVIEW

**Comment by jond01** [comment]
Looks like only one check for python 3.4 failed:
https://travis-ci.org/github/innolitics/dicom-numpy/jobs/683489056
It looks like some installation problem that is not related to this pull request.

File: dicom_numpy/combine_slices.py [hunk 1]
```diff
@@ -233,7 +234,13 @@ def _slice_spacing(slice_datasets):
         slice_positions_diffs = np.diff(sorted(slice_positions))
         return np.mean(slice_positions_diffs)
     else:
-        return 0.0
+        dataset = slice_datasets[0]
+        if hasattr(dataset, 'SliceThickness'):
+            return dataset.SliceThickness
```
Comment (johndgiese) [inline dicom_numpy/combine_slices.py:239]: I think the `SliceThickness` means something different than the `SpacingBetweenSlices`. E.g., for an MR slice, you could sample 1mm worth of tissue yet have the slices be 3mm apart. Thus, I think we should check for the `SpacingBetweenSlices` if present, and if not, return 0.0.

**Comment by jond01** [comment]
Thanks for welcoming me @johndgiese :-)
I will add the unit test for a single slice and the spacing fix, hopefully in the coming week.
Jonathan

File: dicom_numpy/combine_slices.py (hunk 1 above)
Comment (jond01) [reply dicom_numpy/combine_slices.py:239]: You are probably right, I wasn't fully aware of this difference. I will check some data and definitions and fix this accordingly

Comment (jond01) [reply dicom_numpy/combine_slices.py:239]: For future reference, the `SpacingBetweenSlices` is indeed the correct spacing and not `SliceThickness`, as can be checked by the series from here:
https://www.dicomlibrary.com/meddream/?study=1.2.826.0.1.3680043.8.1055.1.20111103111148288.98361414.79379639

**Comment by johndgiese** [comment]
@jond01 let me know once you are ready for another review!

File: tests/test_combine_slices.py [hunk 2]
```diff
@@ -18,6 +18,19 @@ def test_simple_axial_set(self, axial_slices):
...
+        dataset = axial_slices[-1]
+        array, _ = combine_slices([dataset])
+        assert np.array_equal(array, dataset.pixel_array.T[:, :, None])
+
+    def test_single_slice_spacing(self, axial_slices):
+        slice_spacing = 0.65
+        dataset = axial_slices[0]
+        dataset.SpacingBetweenSlices = slice_spacing
+        array, affine = combine_slices([dataset])
+        assert (np.isclose(np.linalg.norm(affine[:, 2]), np.abs(slice_spacing)) and
```
Comment (johndgiese) [inline tests/test_combine_slices.py:31]: Perhaps it would be clearer to split this into two asserts?

Comment (jond01) [reply tests/test_combine_slices.py:31]: Ok

**Comment by jond01** [comment]
Hi, Thanks!
I just discovered some another edge case of RGB dicom slices.
The fix should be very small, so I may add it here or to another PR if you like.

**Comment by johndgiese** [comment]
Thanks for the quick response! Either way works fine. Let me know when you would like me to review. I can then merge it all into a 0.4.0 release.

**Comment by johndgiese** [comment]
Note that @dillonwilliams is working on a small fix as well, so perhaps we can roll that into 0.4.0 too.

**Comment by johndgiese** [comment]
@jond01 how were you thinking of handling RGB slices? I can think of a few possibilities:

- add a 4th dimension to the output array
- return one 3d array for each channel
- flatten the channels into one 3d array

**Comment by jond01** [comment]
I prefer the first option (4d array), to comply with pydicom's pixel array.
The shape of a 3d channelled dicom dataset (multiframe) is:
`ds.pixel_array.T.shape == (n_channels, rows, cols, slices)`

**Comment by johndgiese** [comment]
I agree that this approach is the best. I am a bit surprised that the color will be the first axis, but if that is what DICOM does, it makes sense to use that!

**Comment by johndgiese** [comment]
For the record:

I don't like flattening the channels, since of course this looses information you may want (and there are many ways one can flatten the channels).

I don't like returning multiple arrays, since this would mean the number of `combine_slices`'s return values would vary based on the type of input, which is bad. As it is, the number of axes to the combined array will change, but this seems appropriate.

**Comment by jond01** [comment]
Definitely agree.
I have just finished that, will try to add it here if possible, otherwise in another PR.
Regarding 0.4.0, I think it would be the best to wait some time (<= weeks) and add the other issue fix :)

**Comment by malaterre** [comment]
Just a quick rant. Stop mixing Spacing Between Slices (= Z-Spacing) and Slice Thickness. For reference:

* https://stackoverflow.com/a/15116116/136285

**Comment by jond01** [comment]
No problem ;)
In this PR we concluded that Spacing Between Slices is indeed the correct spacing and not Slice Thickness.

**Comment by johndgiese** [comment]
@malaterre when you say:

> Stop mixing Spacing Between Slices (= Z-Spacing) and Slice Thickness

are you referring to this PR, or something else?
